src/imap.rs

```diff
@@ -1565,47 +1565,49 @@ impl Session {
             .await?
             .map_or(true, |config_token| device_token != config_token);
 
-        if device_token_changed && self.can_metadata() && self.can_push() {
-            let folder = context
-                .get_config(Config::ConfiguredInboxFolder)
-                .await?
-                .context("INBOX is not configured")?;
-
-            let encrypted_device_token =
-                encrypt_device_token(&device_token).context("Failed to encrypt device token")?;
-
-            // We expect that the server supporting `XDELTAPUSH` capability
-            // has non-synchronizing literals support as well:
-            // <https://www.rfc-editor.org/rfc/rfc7888>.
-            let encrypted_device_token_len = encrypted_device_token.len();
-
-            if encrypted_device_token_len <= 4096 {
-                self.run_command_and_check_ok(&format_setmetadata(
-                    &folder,
-                    &encrypted_device_token,
-                ))
-                .await
-                .context("SETMETADATA command failed")?;
+        if self.can_metadata() && self.can_push() {
```

Comment:
as `device_token_changed` is needed only for one if-branch and otherwise also `Config::DeviceToken` set, i find the code easier to understand if   `device_token_changed` is moved inside the `if-branch`:

```
        if self.can_metadata() && self.can_push() {
            let device_token_changed = context
                .get_config(Config::DeviceToken)
                .await?
                .map_or(true, |config_token| device_token != config_token);        
```